src/sentry/integrations/bitbucket_server/integration.py

```diff
@@ -170,6 +170,8 @@ def dispatch(self, request: Request, pipeline) -> Response:
         try:
             request_token = client.get_request_token()
             pipeline.bind_state("request_token", request_token)
+            if not request_token.get("oauth_token"):
+                raise ApiError("Missing oauth_token")
```

Comment:
We shouldn't be raising an exception inside of a `try` block just for control flow.
Also this same bug exists on Jira server so we should fix it there too.

Reply:
Isn't it useful to log the message that we're missing the auth token? How else can we know this going back later?